File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        x_str = str(x)
        n = len(x_str)
        for i in range(n // 2 + 1):
            if x_str[i] != x_str[n-i-1]:
                return False
        return True

File: 0009-palindrome-number/0009-palindrome-number.c
bool isPalindrome(int x)
{
    if (x < 0) {
        return false;
    }

    char x_str[12];
    int n = snprintf(x_str, sizeof(x_str), "%d", x);

    for (int i = 0; i < n / 2; i++) {
        if (x_str[i] != x_str[n - i - 1]) {
            return false;
        }
    }

    return true;
}